src/test/java/bio/terra/app/usermetrics/BardClientTest.java

```diff
@@ -0,0 +1,139 @@
+package bio.terra.app.usermetrics;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.app.configuration.UserMetricsConfiguration;
+import bio.terra.common.category.Unit;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "unittest"})
+@Category(Unit.class)
+@SuppressWarnings("rawtypes")
+public class BardClientTest {
+
+  @SpyBean UserMetricsConfiguration userMetricsConfiguration;
+
+  private AuthenticatedUserRequest user1;
+
+  private AuthenticatedUserRequest user2;
+
+  private String SYNC_PATH;
+
+  private String API_PATH;
+
+  @Before
+  public void setup() {
+    SYNC_PATH = userMetricsConfiguration.getBardBasePath() + "/api/syncProfile";
+    API_PATH = userMetricsConfiguration.getBardBasePath() + "/api/event";
+    user1 =
+        AuthenticatedUserRequest.builder()
+            .setSubjectId("Bob")
+            .setEmail("bob")
+            .setToken("bob token")
+            .build();
+
+    user2 =
+        AuthenticatedUserRequest.builder()
+            .setSubjectId("alice")
+            .setEmail("alice")
+            .setToken("alice token")
+            .build();
+  }
+
+  @Test
+  public void testBardClientLogEvent_happy() {
+    BardClient bardClient = spy(new BardClient(userMetricsConfiguration));
+    RestTemplate restTemplate = mock(RestTemplate.class);
```

Comment:
Suggestion: you could expose these as instance variables, initialize in `setup()`, and only need to set up the `restTemplate` mocking at the top of your tests.

Reply:
I think I'd end up with cache entry bleed across tests if I made only that change, right?  There are ways to deal with that: add and invoke a clearCache in a @BeforeEach; create and use different users between tests; etc.   At that point, I figured I might as well set up a new BardClient per test.   What do you think?

Reply:
Hmm, I thought that the `@Before` annotation on `setup()` indicates that it will be run before each test case, not once at the top of the class:

https://junit.org/junit4/javadoc/4.12/org/junit/Before.html
https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall

For that reason, I don't think that cache bleed would be an issue, but happy to talk more.

Reply:
I recently got tripped up by this as well!  It's definitely not clear.